### PR TITLE
feat(flows): ability to insert cells inline

### LINKED ISSUE #18722: Ability to insert a cell between other cells in a flowbook

### PR DESCRIPTION
**Pull request by alexpaxton**
Closes #18722 

insert-cells-inline

- Render a divider component after each panel which offers an insert menu
- Extend `addCell` function to optionally take an `index` to specify insertion point
- Extend `AddButtons` to optionally take an index and a callback, as well as a required event name for analytics
- Show previous result helper text in all script cells regardless of position
- Add margin at the bottom of the flows list to make resizing the last cell easier
- Add default text to markdown cells
  - If the user adds one and doesn't immediately add text then the cell is blank and strange looking

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by alexpaxton** [comment]
@drdelambre since I used `AddButtons` as a starting point for `InsertCellMenu` there's a lot of repeated code there. I am wondering if it should be extracted into a utility that both components can consume so keeping them in sync isn't an issue down the road

File: ui/src/notebooks/pipes/Query/index.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ register({
     activeQuery: 0,
     queries: [
       {
-        text: '// Write Flux script here',
+        text:
+          '// Write Flux script here\n// use __PREVIOUS_RESULT__ to continue building from the previous cell',
```
Comment (drdelambre) [inline ui/src/notebooks/pipes/Query/index.ts:17]: why this change?

File: ui/src/notebooks/pipes/markdown/index.ts [hunk 2]
```diff
@@ -170,7 +170,7 @@ register({
   component: MarkdownPanel,
   button: 'Markdown',
   initial: () => ({
-    text: TEST_MODE ? TESTCASE : '',
+    text: TEST_MODE ? TESTCASE : 'Type markdown here',
```
Comment (drdelambre) [inline ui/src/notebooks/pipes/markdown/index.ts:173]: do we wanna make this a markdown comment? `[//]: # (Write Markdown below)`

File: ui/src/notebooks/components/panel/InsertCellMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+// Constants
+import {NotebookContext} from 'src/notebooks/context/notebook'
+import {PIPE_DEFINITIONS} from 'src/notebooks'
+
+import {event} from 'src/notebooks/shared/event'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+interface Props {
+  index: number
```
Comment (drdelambre) [inline ui/src/notebooks/components/panel/InsertCellMenu.tsx:22]: could we just make this an optional prop, and then it's the same thing as the addbuttons?

File: ui/src/notebooks/components/panel/InsertCellMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          text={def.button}
+          onClick={() => {
+            let data = def.initial
+            if (typeof data === 'function') {
+              data = data()
+            }
+
+            onInsert()
+
+            event('Notebook Add Button Clicked', {
```
Comment (drdelambre) [inline ui/src/notebooks/components/panel/InsertCellMenu.tsx:57]: should probably change this event name too

**Comment by drdelambre** [comment]
does this also prompt us to drop in an animation library?

**Comment by drdelambre** [comment]
ALSO. if we are unsure of discoverability, we should instrument the hover state (ie: `insert button found` and `insert menu opened`)

File: ui/src/notebooks/pipes/Query/index.ts (hunk 1 above)
Comment (alexpaxton) [reply ui/src/notebooks/pipes/Query/index.ts:17]: I added it in the query registration so it appears on all cells regardless of position. I figure showing it all the time doesn't impact usability

File: ui/src/notebooks/pipes/markdown/index.ts (hunk 2 above)
Comment (alexpaxton) [reply ui/src/notebooks/pipes/markdown/index.ts:173]: I tried the syntax here and it appears as a link, not sure if there is a comment in markdown or nah
